pipenv/environments.py

```diff
@@ -157,7 +157,7 @@ def _is_env_truthy(name):
 https://github.com/sindresorhus/cli-spinners
 """
 
-PIPENV_PIPFILE = os.environ.get("PIPENV_PIPFILE")
+PIPENV_PIPFILE = os.path.expanduser(os.environ.get("PIPENV_PIPFILE")) if os.environ.get("PIPENV_PIPFILE") else None
```

Comment:
IMO a better place for this code is
https://github.com/pypa/pipenv/blob/3b9b7172293169ad5ce0b7be77e6f27e3dbcde7b/pipenv/project.py#L99-L107